src/main/java/com/dsec/backend/DTO/UserHALDTO.java

```diff
@@ -0,0 +1,45 @@
+package com.dsec.backend.DTO;
+
+import javax.validation.Valid;
+import javax.validation.constraints.Email;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Size;
+
+import org.springframework.hateoas.RepresentationModel;
+import org.springframework.hateoas.server.core.Relation;
+
+import lombok.AllArgsConstructor;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+@AllArgsConstructor
+@NoArgsConstructor
+@EqualsAndHashCode(onlyExplicitlyIncluded = true, callSuper = false)
+@Getter
+@Setter
+//@JsonInclude(JsonInclude.Include.NON_NULL)
+@Relation(collectionRelation = "users", itemRelation = "user")
+public class UserHALDTO extends RepresentationModel<UserHALDTO> {
+
+    @NotNull
+    private Integer IdUser;
+
+    @NotNull
+    @Size(min = 1, message = "{Size.UserHALDTO.firstName}")
+    private String firstName;
+
+    @NotNull
+    @Size(message = "{Size.UserHALDTO.lastName}", min = 1)
+    private String lastName;
+
+    @NotNull
+    @Email(message = "{UserHALDTO.email}")
+    private String email;
+
+    @NotNull
+    @Size(message = "{UserHALDTO.password}")
+    private String password;
+
+}
```

Comment:
This is not a valid DTO. It is used for both request and response and contains mixed logic - validation and response serialisation. I can also see that it was created while referring to the project that was given to you as a reference. The difference between this project and the other project is that in the other project the messages are externalised inside the messages.properties file under the ./src/main/resources folder. You should write a message directly because we don't have a need for externalisation yet. I am going to delete this file because the DTO package was renamed to model package and inside it is a DTO that serves the same purpose.